pkg/component/runtime/service.go

```diff
@@ -432,10 +432,44 @@ func (s *ServiceRuntime) uninstall(ctx context.Context) error {
 }
 
 // UninstallService uninstalls the service
-func UninstallService(ctx context.Context, log *logger.Logger, comp component.Component) error {
+func UninstallService(ctx context.Context, log *logger.Logger, comp component.Component, uninstallToken string) error {
+	if comp.InputSpec.Spec.Service.Operations.Uninstall != nil {
+		// Resolve uninstall token parameter
+		comp.InputSpec.Spec.Service.Operations.Uninstall = resolveUninstallTokenArg(comp.InputSpec.Spec.Service.Operations.Uninstall, uninstallToken)
+	}
 	return uninstallService(ctx, log, comp, executeServiceCommand)
 }
 
+//nolint:gosec // was false flagged as hardcoded credentials by linter. it is not.
+const uninstallTokenArg = "--uninstall-token"
+
+// resolveUninstallTokenArg Resolves the uninstall token parameter.
+// If the uninstall spec arguments contains the --uninstall-token then
+// 1. Remove the argument if the value of uninstallToken is empty
+// or
+// 2. Inject the value of uninstallToken after the --uninstall-token argument
+//
+// If args do not contain "--uninstall-token", older endpoint spec, do nothing
+func resolveUninstallTokenArg(uninstallSpec *component.ServiceOperationsCommandSpec, uninstallToken string) *component.ServiceOperationsCommandSpec {
+	if uninstallSpec == nil {
+		return nil
+	}
+
+	spec := *uninstallSpec
+	for i, arg := range spec.Args {
+		if arg == uninstallTokenArg {
+			if uninstallToken == "" { // Remove --uninstall-token argument if the token is empty
+				spec.Args = append(spec.Args[:i], spec.Args[i+1:]...)
```

Comment:
I think you should copy the args into a new array so that you are not affecting the actual spec. Line 458 above is copying the spec structure but the array is only going to be a pointer copy.